index.html

```
@@ -927,16 +958,91 @@
-      environment, a payment handler should <em>not</em> open a
-      window in a new browser tab, as this is too dissociated from
-      the checkout context.</p>
-      <p>User agents SHOULD display the origin of a running payment
-      handler to the user.</p>
+      <p>A payment handler that requires visual display and user
+      interaction, may call <a>openWindow()</a>. Since user agents
+      know that this method is connected to the payment request
+      event, they SHOULD render the window in a way that is
+      consistent with the flow and not confusing to the user.</p>
+      <p>The resulting window client is bound to the tab/window
+      that initiated the <a>payment request</a>. A single
+      <a>payment handler</a> SHOULD NOT be allowed to open more
+      than one client window using this method.</p>
+      <section>
+        <h3><dfn>Open Window Algorithm</dfn></h3>
+        <p class="note">This algorithm is <i>very</i> similar to
+        the <a href=
+        "https://www.w3.org/TR/service-workers/#clients-openwindow">
+        Open Window Algorithm</a> in the Service Workers
+        specification. Should we refer to the Service Workers
+        specification instead of copying their steps?</p>
+        <ol class="algorithm">
+          <li>Let <var>event</var> be this
+          <a>PaymentRequestEvent</a>.
+          </li>
+          <li>Let <var>request</var> be the <a>PaymentRequest</a>
+          that triggered this payment request event.
+          </li>
+          <li>If <var>request</var>.[[\state]] is not
+          "interactive", then return a Promise rejected with a <a>
+            DOMException</a> whose value is
+            "<a>InvalidStateError</a>".
+          </li>
+          <li>Let <var>url</var> be the result of parsing the
+          <var>url</var> argument.</li>
+          <li>If <var>url</var> is failure, return a Promise
+          rejected with a <a>TypeError</a>.
```

You should re-throw the error instead of creating an empty one.